app/serializers/serializer.rb

```diff
@@ -2,14 +2,23 @@
 
 class Serializer
   def self.for(model:, version:)
-    Object.const_get("#{module_name_for(model)}::#{class_name_for(version)}")
+    module_name = module_name_for(model)
+    constant_module = Object.const_get(module_name)
+    class_name = class_name_for(version)
+    if constant_module.const_defined?(class_name)
+      constant_module.const_get(class_name)
+    else
+      raise(NotImplementedError, "#{module_name}::#{class_name}")
```

Comment:
Yeah I was thinking it be the direction going forward. Since these are a factory method of sort that returns a `versioned` item.

If the requested versioned item don't exist then we can safely say it that api `version` as a sum is incomplete.

Reply:
Carded ashttps://trello.com/c/iQnQ1jfH/1821-guard-access-to-versioned-api-items i leave it to